File: codex/js/Strings/Replace and capture.js
// js, strings, replace, capture, regex, case-insensitive regex, RegExp


const str = "Hello";

str.replace(/^(.{4}).+$/, "$1");    // 'Hell'
str.replace(new RegExp("^(.{4}).+$"), "$1");
str.replace(/l/, 'L');  // 'HeLlo'
str.replaceAll(/l/g, 'L');  // 'HeLLo' (replaceAll needs global flag)
str.replaceAll(/L/ig, '*');  // 'He**o' (case insensitive)
// named capture groups
str.replace(/(?<doublels>l{2})/g, "_$<doublels>_")

File: codex/r/Strings/Replace and capture.R
# r, strings, replace, capture, regex, str_replace, str_replace_all, case-insensitive regex

str <- "Hello"
str_replace(str, "^(.{4}).+$", "\\1")   # "Hell"
str_replace(str, "l", "L")              # "HeLlo"
str_replace_all(str, "l", "L")          # "HeLLo"
str_replace_all(str, "(?i)L", "*")      # "He**o"
